### PR TITLE
fix: improve shared drives UI/UX (VO-651) (VO-652) (VO-653)

### PR DESCRIPTION
**Pull request by acezard**
remove url suffix
add ellipsis

image

### CODE REVIEW

File: src/modules/filelist/File.jsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ const File = ({
   onFolderOpen,
   onFileOpen,
   disableSelection = false,
-  isExternalDrive
+  isSharedDrives
```
Comment (cballevre) [inline src/modules/filelist/File.jsx:50]: Why rename external by shared? As far as I'm concerned, the two concepts are similar but different. For example, we'll have a different icon for displaying shared drives in the tabs view

Comment (acezard) [reply src/modules/filelist/File.jsx:50]: it was to match the doctype `io.cozy.files.shared-drives-dir`, what do you think?

Comment (cballevre) [reply src/modules/filelist/File.jsx:50]: I understand your change better, but I'd put more nuance into it. I would keep the external naming for everything related to the external link. For the moment, we only retrieve external links via the magic folder but soon we'll have folders for shared drives with a special design.

File: src/locales/en.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     "item_settings": "Settings",
     "item_collect": "Administrative",
     "item_external_drives": "Shared Drives",
```
Comment (cballevre) [inline src/locales/en.json:11]: I think this should be `item_shared_drives` to be consistent

File: src/modules/filelist/File.jsx (hunk 1 above)
Comment (cballevre) [reply src/modules/filelist/File.jsx:50]: For example, I will still use the external here

File: src/modules/navigation/NavContent.jsx [hunk 3]
```diff
@@ -6,9 +6,10 @@ import Circle from 'cozy-ui/transpiled/react/Circle'
 import Icon from 'cozy-ui/transpiled/react/Icon'
 import OpenWithIcon from 'cozy-ui/transpiled/react/Icons/Openwith'
 import { NavIcon, NavText } from 'cozy-ui/transpiled/react/Nav'
+import Typography from 'cozy-ui/transpiled/react/Typography'
 import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'
 
-const NavContent = ({ icon, badgeContent, label, external }) => {
+const NavContent = ({ icon, badgeContent, label, sharedDrives }) => {
```
Comment (cballevre) [inline src/modules/navigation/NavContent.jsx:12]: Here I will also still use external as this determines whether the external icon appears. But I prefer this if to separate cases 👍

File: src/modules/navigation/NavItem.jsx [hunk 4]
```diff
@@ -31,21 +30,20 @@ const NavItem = ({
   rx,
   clickState,
   badgeContent,
-  external,
+  sharedDrives,
```
Comment (cballevre) [inline src/modules/navigation/NavItem.jsx:33]: I would keep `external` also here

File: src/modules/views/Folder/SharedDrives.tsx [hunk 5]
```diff
@@ -51,7 +48,7 @@ export const ExtraDrive = ({
                 onFileOpen={handleFileOpen}
                 isInSyncFromSharing={false}
                 disableSelection={true}
-                isExternalDrive={true}
+                isSharedDrives={true}
```
Comment (cballevre) [inline src/modules/views/Folder/SharedDrives.tsx:51]: This is where the subtlety comes in. This component knows that it is displaying the contents of shared drives. But we specify the display cases for the children. Here an external drive and in the future a shared drive.

File: src/locales/en.json (hunk 2 above)
Comment (acezard) [reply src/locales/en.json:11]: indeed, seems I missed it

File: src/modules/filelist/File.jsx (hunk 1 above)
Comment (acezard) [reply src/modules/filelist/File.jsx:50]: agreed

File: src/modules/navigation/NavContent.jsx (hunk 3 above)
Comment (acezard) [reply src/modules/navigation/NavContent.jsx:12]: agreed

File: src/modules/views/Folder/SharedDrives.tsx (hunk 5 above)
Comment (acezard) [reply src/modules/views/Folder/SharedDrives.tsx:51]: yes, instead I will check if it's of type shortcut. I assume shared drives will be of type file or directory?

Comment (cballevre) [reply src/modules/views/Folder/SharedDrives.tsx]: From my understanding it would be a directory

File: src/modules/navigation/NavItem.jsx [hunk 6]
```diff
@@ -31,21 +30,20 @@ const NavItem = ({
...
   return (
-    <UINavItem secondary={secondary} onClick={onClick}>
+    <UINavItem secondary={secondary}>
       <NavLink to={to} rx={rx} clickState={clickState}>
         <NavContent
           icon={icon}
           label={forcedLabel ?? t(`Nav.item_${label}`)}
           badgeContent={badgeContent}
-          external={external}
+          sharedDrives={externalDrive}
```
Comment (cballevre) [inline src/modules/navigation/NavItem.jsx:46]: I think there's a renaming error here

File: src/modules/filelist/File.jsx [hunk 7]
```diff
@@ -47,7 +47,7 @@ const File = ({
   onFolderOpen,
   onFileOpen,
   disableSelection = false,
-  isExternalDrive
+  externalDrive
```
Comment (cballevre) [inline src/modules/filelist/File.jsx:50]: Usual I prefer to prefix with an `is` the boolean but the props just below don't follow this convention so its only a nit

File: src/modules/navigation/NavContent.jsx [hunk 8]
```diff
@@ -6,9 +6,10 @@ import Circle from 'cozy-ui/transpiled/react/Circle'
 import Icon from 'cozy-ui/transpiled/react/Icon'
 import OpenWithIcon from 'cozy-ui/transpiled/react/Icons/Openwith'
 import { NavIcon, NavText } from 'cozy-ui/transpiled/react/Nav'
+import Typography from 'cozy-ui/transpiled/react/Typography'
 import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'
 
-const NavContent = ({ icon, badgeContent, label, external }) => {
+const NavContent = ({ icon, badgeContent, label, externalDrive }) => {
```
Comment (cballevre) [inline src/modules/navigation/NavContent.jsx:12]: nit: `external` is more general maybe we can use it for other purpose in the futur

**Review by cballevre — APPROVED** [review]
LTGM 👍  after fixing the renaming error

File: src/modules/navigation/NavContent.jsx (hunk 8 above)
Comment (Crash--) [reply src/modules/navigation/NavContent.jsx:12]: Capture d’écran 2024-04-16 à 15 37 07
from https://github.com/cozy/cozy-guidelines/?tab=readme-ov-file#naming-of-functions ;)

**Pull request merged**